src/ModelObjects/representations.rs

```diff
@@ -125,53 +133,78 @@ impl<'a> SystemRepresentation {
+                let mut left = vec![];
+                let mut right = vec![];
+                let mut success = left_side
+                    .collect_open_transitions(locations, index, action, &mut left, sync_type);
+
+                success = success
                     || right_side
-                        .collect_open_edges(locations, index, action, open_edges, sync_type)
+                        .collect_open_transitions(locations, index, action, &mut right, sync_type);
+
+                // Independent actions
+                if left.is_empty() || right.is_empty() {
+                    open_transitions.append(&mut left);
+                    open_transitions.append(&mut right);
+                }
+                // Synchronized actions
+                else {
+                    open_transitions.append(&mut Transition::combinations(&mut left, &mut right));
+                }
+
+                success
             }
             SystemRepresentation::Conjunction(left_side, right_side) => {
-                let open_edges_len = open_edges.len();
-                if left_side.collect_open_edges(locations, index, action, open_edges, sync_type) {
-                    let left_found_transitions = open_edges_len != open_edges.len();
-                    if right_side
-                        .collect_open_edges(locations, index, action, open_edges, sync_type)
-                    {
-                        let right_found_transitions = open_edges_len != open_edges.len();
-                        return left_found_transitions == right_found_transitions;
-                    }
-                }
-                false
+                let mut left = vec![];
+                let mut right = vec![];
+                let mut success = left_side
+                    .collect_open_transitions(locations, index, action, &mut left, sync_type);
+
+                success = success
+                    && right_side
+                        .collect_open_transitions(locations, index, action, &mut right, sync_type);
+
+                open_transitions.append(&mut Transition::combinations(&mut left, &mut right));
+
+                success
             }
             SystemRepresentation::Parentheses(rep) => {
-                rep.collect_open_edges(locations, index, action, open_edges, sync_type)
+                rep.collect_open_transitions(locations, index, action, open_transitions, sync_type)
             }
             SystemRepresentation::Component(comp) => {
                 let next_edges =
                     comp.get_next_edges(locations[*index].get_location(), action, *sync_type);
-
-                if !next_edges.is_empty() {
-                    open_edges.push((comp, next_edges, *index));
+                for e in next_edges {
+                    open_transitions.push(Transition {
+                        edges: vec![(comp, e, *index)],
+                    });
```

Comment:
I thought we were only supposed to collect open transitions. But using `get_next_edges` gets both open and closed edges.